packages/next/build/webpack/loaders/next-swc-loader.js

```diff
@@ -29,12 +29,16 @@ DEALINGS IN THE SOFTWARE.
 import { getOptions } from 'next/dist/compiled/loader-utils'
 import { transform } from '../../swc'
 
+const nextDistPath =
+  /(next[\\/]dist[\\/]shared[\\/]lib)|(next[\\/]dist[\\/]client)|(next[\\/]dist[\\/]pages)/
```

Comment:
Are you missing the `^` for "starts with"?

Reply:
No because this is a matches that matches further down the path, it's the same regex used in our babel-loader currently.